src/Resources/config/form_types.php

```diff
@@ -0,0 +1,151 @@
+
+return static function (ContainerConfigurator $containerConfigurator): void {
+    // Use "service" function for creating references to services when dropping support for Symfony 4.4
+    // Use "param" function for creating references to parameters when dropping support for Symfony 5.1
+    $containerConfigurator->services()
+
+        ->set('sonata.admin.form.type.admin', AdminType::class)
+            ->public()
+            ->tag('form.type', ['alias' => 'sonata_type_admin'])
+            ->args([
+                new ReferenceConfigurator('sonata.admin.helper'),
+            ])
+
+        ->set('sonata.admin.form.type.model_choice', ModelType::class)
+            ->public()
+            ->tag('form.type', ['alias' => 'sonata_type_model'])
+            ->args([
+                new ReferenceConfigurator('property_accessor'),
+            ])
+
+        ->set('sonata.admin.form.type.model_list', ModelListType::class)
+            ->public()
+            ->tag('form.type', ['alias' => 'sonata_type_model_list'])
+
+        ->set('sonata.admin.form.type.model_reference', ModelReferenceType::class)
+            ->public()
+            ->tag('form.type', ['alias' => 'sonata_type_model_reference'])
+
+        ->set('sonata.admin.form.type.model_hidden', ModelHiddenType::class)
+            ->public()
+            ->tag('form.type', ['alias' => 'sonata_type_model_hidden'])
+
+        ->set('sonata.admin.form.type.model_autocomplete', ModelAutocompleteType::class)
+            ->public()
+            ->tag('form.type', ['alias' => 'sonata_type_model_autocomplete'])
+
+        ->set('sonata.admin.form.type.collection', CollectionType::class)
+            ->public()
+            ->tag('form.type', ['alias' => 'sonata_type_native_collection'])
+
+        ->set('sonata.admin.doctrine_orm.form.type.choice_field_mask', ChoiceFieldMaskType::class)
+            ->public()
+            ->tag('form.type', ['alias' => 'sonata_type_choice_field_mask'])
+
+        ->set('sonata.admin.form.extension.field', FormTypeFieldExtension::class)
+            ->public()
+            ->tag('form.type_extension', [
+                'alias' => 'form',
+                'extended_type' => FormType::class,
+                'extended-type' => FormType::class,
+            ])
+            ->args(['', ''])
+
+        ->set('sonata.admin.form.extension.field.mopa', MopaCompatibilityTypeFieldExtension::class)
+            ->public()
+            ->tag('form.type_extension', [
+                'alias' => 'form',
+                'extended_type' => FormType::class,
+                'extended-type' => FormType::class,
+            ])
+
+        ->set('sonata.admin.form.extension.choice', ChoiceTypeExtension::class)
+            ->public()
+            ->tag('form.type_extension', [
+                'alias' => 'choice',
+                'extended_type' => SymfonyChoiceType::class,
+                'extended-type' => ChoiceType::class,
+            ])
```

Comment:
the double extended_type looks strange

Reply:
Indeed, apparently `extended-type` only works with XML configuration.